rest/system_metrics_routes.go

```diff
@@ -40,16 +43,23 @@ func (h *systemMetricsGetByTypeHandler) Factory() gimlet.RouteHandler {
 func (h *systemMetricsGetByTypeHandler) Parse(_ context.Context, r *http.Request) error {
 	var err error
 
-	h.opts.TaskID = gimlet.GetVars(r)["task_id"]
-	h.metricType = gimlet.GetVars(r)["type"]
+	h.findOpts.TaskID = gimlet.GetVars(r)["task_id"]
+	h.downloadOpts.MetricType = gimlet.GetVars(r)["type"]
 	vals := r.URL.Query()
+	h.downloadOpts.PageSize = softSizeLimit
```

Comment:
Should users be able to specify this in their request?

Reply:
I don't think so, I don't want to trust users to set reasonable page sizes. We already set it for buildlogger and it seems to work fine.